app/models/ncr/work_order.rb

```diff
@@ -121,13 +121,23 @@ def approving_official
       self.approvers.first
     end
 
-    def current_approver_email_address
-      if self.pending?
-        self.currently_awaiting_approvers.first.email_address
-      elsif self.approving_official
-        self.approving_official.email_address
+    def current_approver
+      if pending?
+        currently_awaiting_approvers.first
+      elsif approving_official
+        approving_official
+      elsif emergency and approvers.empty?
+        nil
       else
-        self.system_approver_emails.first
+        User.for_email(self.system_approver_emails.first)
+      end
+    end
+
+    def final_approver
+      if !emergency and approvers.any?
+        approvers.last
+      else
```

Comment:
don't need this `else` -- I know we discussed this in a previous PR with @afeld but it seems to be that if we want to always do an `else nil` on a conditional in Ruby (which is the behavior whether you do it explicitly or not) then we should do it everywhere and not just selectively. I would vote not to do it.